### PR TITLE
[Merged] Mocha to jest

### PR DESCRIPTION
**Pull request by McCloudFace**
Plz don't merge, review only plz and TY!

@weber93 - Primary
@cjm771 - Secondary

### CODE REVIEW

**Comment by mweber-ak** [comment]
@McCloudFace I'll jump on this tonight/tomorrow AM

**Comment by cjm771** [comment]
looks like your feature branch is behind the master, try rebasing it to remove and resolve these conflicts. 

``` sh
$> git checkout master
$> git pull
$> git checkout mocha-to-jest
$> git rebase master
```

**Review by mweber-ak — COMMENTED** [review]
@McCloudFace 

Your data appears to come back in the following format
```
{
  "data": {
    "attributes": {
      "trail_name": "Golden Gate Park Trail",
      "distance": 6,
      "distance_units": "miles",
      "elevation_gain": 351,
      "elevation_units": "ft",
      "description": "Golden Gate Park Trail is a Bernie favorite. A 6.1 mile heavily trafficked loop trail located near San Francisco, California that features a lake and is good for all skill levels. The trail offers a number of activity options and is accessible year-round. Dogs are also able to use this trail but must be approved by Bernie.",
      "route_type": "Loop",
      "difficulty": "Easy",
      "general_area": "Golden Gate Park",
      "origin": "https://www.alltrails.com/trail/us/california/golden-gate-park-trail",
      "tags": ["dogs on leash", "hiking", "mountain biking", "trail running", "walking", "views", "wildlife", "muddy", "dog friendly", "backpacking", "birding", "historic site"]
    },
    "trail_id": 1,
    "type": "trail"
  }
}
```

But in your test on line 10 on db.test.js, it accesses all the properties as if they were all on the same "level" of the object. Example. `trail_name` is accessed the same way as `trail_id` but unless your data returns differently than shown above, it would be under `trailResult[0].attributes`. Hopefully, that makes sense. I'm sure I am missing something that clears that. Just let me know.

**Comment by McCloudFace** [comment]
@cjm771  Thanks! Rebase should be done & pushed

File: test/db.test.js [hunk 1]
```diff
@@ -1,48 +1,34 @@
...
 
-      db.getAtrail(1, function (trailResult) {
-        expect(trailResult[0])
-          .to.be.a('object')
-          .and.to.have.all.keys('trail_id', 'trail_name', 'distance', 'distance_units', 'elevation_gain', 'elevation_units', 'description', 'route_type', 'difficulty', 'general_area', 'origin');
-        done();
-      });
-    });
+  db.getAtrail(1, function (trailResult) {
+    expect(trailResult[0]).toBeInstanceOf(Object);
```
Comment (cjm771) [inline test/db.test.js:11]: this is fine , and knitpickyness, but i would maybe simplify this with an array of properties and a `foreach` over to check what you need to check. But just a comment no change needed

``` js
['trail_id', 'trail_name', ...].forEach((prop) => { expect(trailResult[0]).toHaveProperty(prop) })
```

**Review by cjm771 — COMMENTED** [review]
Looks good tests pass, see minor comment

File: test/db.test.js (hunk 1 above)
Comment (McCloudFace) [reply test/db.test.js:11]: Great tip, thanks. That would be a better test of the results object, thanks. I'll add an issue to my repo to refactor this test. Thanks!

**Comment by McCloudFace** [comment]
@weber93 Good call out, thanks. Added an issue to refactor this in a new ticket.

@weber93 or @cjm771 If you could merge this one when you get a sec, I'd appreciate it. And thanks very much for the reviews, much appreciated.

**Pull request merged**